core/client/src/main/java/alluxio/client/lineage/LineageMasterClient.java

```diff
@@ -118,17 +119,19 @@ public Boolean call() throws AlluxioTException, TException {
    * @param path the path to the file
    * @param blockSizeBytes the size of the block in bytes
    * @param ttl the time to live for the file
+   * @param ttlExpiryAction Action to take after Ttl expiry
    * @return the value of the lineage creation result
    * @throws IOException if a non-Alluxio exception occurs
    * @throws LineageDoesNotExistException if the file does not exist
    * @throws AlluxioException if an Alluxio exception occurs
    */
   public synchronized long reinitializeFile(final String path, final long blockSizeBytes,
-      final long ttl) throws IOException, LineageDoesNotExistException, AlluxioException {
+      final long ttl, final TtlExpiryAction ttlExpiryAction)
```

Comment:
this method should take non-thrift Ttl action and encode it as thrift ... the clients should be the only layer aware of the fact that the underlying protocol is Thrift